content/ADR-175-remove-assets-from-build-local-addressables.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: adr
+adr: 175
+title: Remove assets from Renderer's build. Addressables Integration
+date: 2023-01-16
+status: Review
+type: Standards Track
+spdx-license: CC0-1.0
+authors:
+  - ajimenezDCL
+  - mikhail-dcl
+---
+
+## Abstract
+This document describes an approach on how to make the reference's client lightweight by streaming the currently embedded assets. 
+
+## Problem
+Every nice UI, sound effect, particle or material we add to the renderer increases its size. This directly impact loading times or caching since the client's size will go above the max allowed for it. The idea is to pack textures, audio files and other assets, and stream them in runtime. 
+
+## Solution
+### Pack the Assets
+Similar to what we are doing with scene assets, we can pack everything into one or multiple (to always be under the cache size limit) asset packs to be consumed by the client. This solution should be part of a different system than the one used for scene asset.
+
+### [Addressables](https://docs.unity3d.com/Packages/com.unity.addressables@1.21/manual/index.html)
+The Addressables system provides tools and scripts to organize and package content for an application and an API to load and release assets at runtime.
+It's an additional abstraction layer over the Asset Bundles System but it's not limited to this. "Addressables" gives the flexibility to adjust where to host assets and provides mechanisms for dependency resolution, reference counting, simulation, validation, and diagnostics out of the box.
+
+### Strategy
+In the first iteration we are going to use the "Local" mode only:
+  - the default addressables profile is suitable for this purpose, we don't have to introduce our own
+<img alt="image" src="https://user-images.githubusercontent.com/118179774/214870799-4fe1b14c-9b17-4485-8958-5ed11eeafdd1.png">
+
+  - the remote section of the profile we are going to ignore
+  - the local section refers to building and loading assets from the "Streaming Assets" directory
+  - on WebGL "Streaming Assets" are hosted in the same storage where the main binary (wasm) is located, and, thus, are loaded via "Web Request"
```

Comment:
I think its important to add where and how the StreamingAssets are definied in the config file. I had to modify it to use it in the cjk fonts; otherwise the default did not work. And it can be pretty hidden if you dont know where to look.

https://github.com/decentraland/unity-renderer/blob/4b383bd926143197baccf0a94dd5a501b61b5b74/browser-interface/src/index.ts#L97

Reply:
Can we add this link along with some explanations of how it works?

Reply:
Done